parser/lexer.go

```diff
@@ -285,70 +285,107 @@ func scanQuotedIdent(s *Scanner) (tok int, pos Pos, lit string) {
 }
 
 func startString(s *Scanner) (tok int, pos Pos, lit string) {
-	return s.scanString()
+	tok, pos, lit = s.scanString()
+
+	// Quoted strings placed next to each other are concatenated to a single string.
+	// See http://dev.mysql.com/doc/refman/5.7/en/string-literals.html
+	ch := s.skipWhitespace()
+	for ch == '\'' || ch == '"' {
+		_, _, lit1 := s.scanString()
+		lit = lit + lit1
+		ch = s.skipWhitespace()
+	}
+	return
+}
+
+// lazyBuf is used to avoid buf if possible.
+type lazyBuf struct {
+	flag bool
```

Comment:
avoid bytes.Buffer.String().
when bytes.Buffer.String() method is called, a new string object is allocated.
I use the bytes.Buffer to deal with escape, such as \r \n \t... if no escape exists, buff is not necessary, use str[i:j] is enough